### PR TITLE
Run the legacy test suite on windows

### PR DESCRIPTION
**Pull request by krocard**
- Adapt the legacy test suite to run on windows
    + no longer use the shell to span commands
    + no longer use cat to read files
    + use os.environ to retrieve environment variables, not `$variable_name` in a command
 - legacy test are more than twice as fast (6s run on travis) !
 - found some bugs in the tests. Corrected some, left most (search for fixmes)
 - corrected some nonsense (like using a file to split a string). Not all though, it's legacy after all.
 - fixe bugs in test-platform and remote-process
 - disable coverage on clang (was not working and making the executable segfaulting on exit)
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-143168183%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-143168260%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-143801897%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-145091304%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41045176%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41045184%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41045360%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41123716%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41128152%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41128163%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-145499408%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-145499842%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23issuecomment-143168183%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Close%20to%20avoid%20triggering%20useless%20appveyor%20builds.%22%2C%20%22created_at%22%3A%20%222015-09-25T09%3A21%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Will%20reopen%20when%20ready%20to%20review.%22%2C%20%22created_at%22%3A%20%222015-09-25T09%3A22%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ready%20to%20be%20tested%20by%20appveyor.%22%2C%20%22created_at%22%3A%20%222015-09-28T16%3A50%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%40OznOg%20%40dawagner%20%40cc6565%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-02T17%3A22%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%20but%20great%20job%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-10-05T11%3A21%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22Please%20resubmit%20against%20the%20master%20branch.%22%2C%20%22created_at%22%3A%20%222015-10-05T11%3A22%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20c6a45fe69c4228a95112dde0842e24dea2eebf2e%20test/functional-tests-legacy/Util/PfwUnitTestLib.py%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41128163%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22succeed%22%2C%20%22created_at%22%3A%20%222015-10-05T10%3A09%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests-legacy/Util/PfwUnitTestLib.py%3AL28-108%22%7D%2C%20%22Pull%201afc1078e9dc25d78c948a0f4e1678b0834d6709%20remote-processor/CMakeLists.txt%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41045176%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20comment%22%2C%20%22created_at%22%3A%20%222015-10-02T17%3A23%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/CMakeLists.txt%3AL49-56%22%7D%2C%20%22Pull%201afc1078e9dc25d78c948a0f4e1678b0834d6709%20test/functional-tests-legacy/Util/PfwUnitTestLib.py%2070%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41045360%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22an%20implementation%22%2C%20%22created_at%22%3A%20%222015-10-02T17%3A25%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests-legacy/Util/PfwUnitTestLib.py%3AL28-101%22%7D%2C%20%22Pull%201afc1078e9dc25d78c948a0f4e1678b0834d6709%20ctest/CMakeLists.txt%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41045184%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20comment%22%2C%20%22created_at%22%3A%20%222015-10-02T17%3A23%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ctest/CMakeLists.txt%3AL72-79%22%7D%2C%20%22Pull%20c6a45fe69c4228a95112dde0842e24dea2eebf2e%20test/functional-tests-legacy/Util/PfwUnitTestLib.py%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41128152%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22its%22%2C%20%22created_at%22%3A%20%222015-10-05T10%3A09%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests-legacy/Util/PfwUnitTestLib.py%3AL28-108%22%7D%2C%20%22Pull%20c6a45fe69c4228a95112dde0842e24dea2eebf2e%20test/functional-tests-legacy/PfwTestCase/Domains/tDomain_Configuration_Selection.py%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/250%23discussion_r41123716%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20comment%20can%20be%20removed.%22%2C%20%22created_at%22%3A%20%222015-10-05T09%3A12%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/functional-tests-legacy/PfwTestCase/Domains/tDomain_Configuration_Selection.py%3AL111-117%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#issuecomment-143168183'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> Close to avoid triggering useless appveyor builds.
- <a href='https://github.com/krocard'></a> Will reopen when ready to review.
- <a href='https://github.com/krocard'></a> Ready to be tested by appveyor.
- <a href='https://github.com/krocard'></a> :+1: @OznOg @dawagner @cc6565 please review
- <a href='https://github.com/dawagner'></a> :-1: but great job :)
- <a href='https://github.com/dawagner'></a> Please resubmit against the master branch.
- [x] <a href='#crh-comment-Pull 1afc1078e9dc25d78c948a0f4e1678b0834d6709 remote-processor/CMakeLists.txt 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#discussion_r41045176'>File: remote-processor/CMakeLists.txt:L49-56</a></b>
- <a href='https://github.com/krocard'></a> Add comment
- [x] <a href='#crh-comment-Pull 1afc1078e9dc25d78c948a0f4e1678b0834d6709 ctest/CMakeLists.txt 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#discussion_r41045184'>File: ctest/CMakeLists.txt:L72-79</a></b>
- <a href='https://github.com/krocard'></a> Add comment
- [x] <a href='#crh-comment-Pull 1afc1078e9dc25d78c948a0f4e1678b0834d6709 test/functional-tests-legacy/Util/PfwUnitTestLib.py 70'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#discussion_r41045360'>File: test/functional-tests-legacy/Util/PfwUnitTestLib.py:L28-101</a></b>
- <a href='https://github.com/krocard'></a> an implementation
- [ ] <a href='#crh-comment-Pull c6a45fe69c4228a95112dde0842e24dea2eebf2e test/functional-tests-legacy/PfwTestCase/Domains/tDomain_Configuration_Selection.py 22'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#discussion_r41123716'>File: test/functional-tests-legacy/PfwTestCase/Domains/tDomain_Configuration_Selection.py:L111-117</a></b>
- <a href='https://github.com/dawagner'></a> This comment can be removed.
- [ ] <a href='#crh-comment-Pull c6a45fe69c4228a95112dde0842e24dea2eebf2e test/functional-tests-legacy/Util/PfwUnitTestLib.py 19'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#discussion_r41128152'>File: test/functional-tests-legacy/Util/PfwUnitTestLib.py:L28-108</a></b>
- <a href='https://github.com/dawagner'></a> its
- [ ] <a href='#crh-comment-Pull c6a45fe69c4228a95112dde0842e24dea2eebf2e test/functional-tests-legacy/Util/PfwUnitTestLib.py 20'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/250#discussion_r41128163'>File: test/functional-tests-legacy/Util/PfwUnitTestLib.py:L28-108</a></b>
- <a href='https://github.com/dawagner'></a> succeed

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/250?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/250?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/250'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by krocard** [comment]
Close to avoid triggering useless appveyor builds.

**Comment by krocard** [comment]
Will reopen when ready to review.

**Comment by krocard** [comment]
Ready to be tested by appveyor.

**Comment by krocard** [comment]
:+1: @OznOg @dawagner @cc6565 please review

File: remote-processor/CMakeLists.txt [hunk 1]
```diff
@@ -49,7 +49,7 @@ target_include_directories(remote-processor PUBLIC "${ASIO_DIR}")
 include_directories("${PROJECT_SOURCE_DIR}/utility"
                     "${CMAKE_CURRENT_BINARY_DIR}")
 
-target_compile_definitions(remote-processor PUBLIC ASIO_STANDALONE)
+target_compile_definitions(remote-processor PUBLIC ASIO_STANDALONE ASIO_HAS_STD_CHRONO)
```
Comment (krocard) [inline remote-processor/CMakeLists.txt]: Add comment

File: ctest/CMakeLists.txt [hunk 2]
```diff
@@ -72,7 +72,7 @@ function(set_test_env TestName)
 
     # Quotes are necessary. Otherwise `;` escapes are discarded
     set_property(TEST ${TestName} PROPERTY ENVIRONMENT
-                 "PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}${SEP}${TEST_PATH}"
+                 "PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}${SEP}${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/$<CONFIGURATION>${SEP}${TEST_PATH}"
```
Comment (krocard) [inline ctest/CMakeLists.txt]: Add comment

File: test/functional-tests-legacy/Util/PfwUnitTestLib.py [hunk 3]
```diff
@@ -28,43 +28,73 @@
...
 
     # Starts the HAL exe
     def startHal(self):
-        cmd= "test-platform $PFW_TEST_CONFIGURATION"
-        subprocess.Popen(cmd, shell=True)
-        pass
+        cmd= ["test-platform", os.environ["PFW_TEST_CONFIGURATION"], self.testPlatformPort]
+        self.setRemoteProcess(subprocess.Popen(cmd))
+        # Wait for the test-platform listening socket
+        # A while loop trying to bind to the socket may be a implementation:
```
Comment (krocard) [inline test/functional-tests-legacy/Util/PfwUnitTestLib.py]: an implementation

File: test/functional-tests-legacy/PfwTestCase/Domains/tDomain_Configuration_Selection.py [hunk 4]
```diff
@@ -112,16 +111,6 @@ def test_Combinatorial_Criteria(self):
         log.I("Command [listDomains] : correctly executed")
 
         # Domains listing backup
```
Comment (dawagner) [inline test/functional-tests-legacy/PfwTestCase/Domains/tDomain_Configuration_Selection.py]: This comment can be removed.

File: test/functional-tests-legacy/Util/PfwUnitTestLib.py [hunk 5]
```diff
@@ -28,43 +28,80 @@
...
 class RemoteCli(object):
-    def sendCmd(self, cmd, *args):
-        shell_cmd = " ".join([self.platform_command, cmd])
+
+    def setRemoteProcess(self, remoteProcess):
+        self.remoteProcess = remoteProcess
+
+    def sendCmd(self, cmd, *args, **kwargs):
+        """ Execute a remote-process command and assert it's result.
+            @param[in] cmd, *args the command to execute and it's arguments
```
Comment (dawagner) [inline test/functional-tests-legacy/Util/PfwUnitTestLib.py]: its

File: test/functional-tests-legacy/Util/PfwUnitTestLib.py [hunk 6]
```diff
@@ -28,43 +28,80 @@
...
+            @param[in] expectSuccess If True, assert that the command will success
```
Comment (dawagner) [inline test/functional-tests-legacy/Util/PfwUnitTestLib.py]: succeed

**Comment by dawagner** [comment]
:-1: but great job :)

**Comment by dawagner** [comment]
Please resubmit against the master branch.

File: test/functional-tests-legacy/Util/PfwUnitTestLib.py (hunk 6 above)
Comment (krocard) [inline test/functional-tests-legacy/Util/PfwUnitTestLib.py]: :+1:<a href='#crh-update-none'></a>

**Comment by krocard** [comment]
:+1:<a href='#crh-update-none'></a>

File: remote-processor/CMakeLists.txt (hunk 1 above)
Comment (krocard) [inline remote-processor/CMakeLists.txt]: :+1:<a href='#crh-update-none'></a>

File: test/functional-tests-legacy/Util/PfwUnitTestLib.py (hunk 3 above)
Comment (krocard) [inline test/functional-tests-legacy/Util/PfwUnitTestLib.py]: :+1:<a href='#crh-update-none'></a>

File: ctest/CMakeLists.txt (hunk 2 above)
Comment (krocard) [inline ctest/CMakeLists.txt]: :+1:<a href='#crh-update-none'></a>

File: test/functional-tests-legacy/Util/PfwUnitTestLib.py (hunk 5 above)
Comment (krocard) [inline test/functional-tests-legacy/Util/PfwUnitTestLib.py]: :+1:<a href='#crh-update-complete'></a>

**Pull request closed without merging**